### PR TITLE
Feature/change authority

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/70885

This PR lays the base for a reusable change authority with tests. Policies are set by classes implementing the `Tribe__Change_Authority__Interface` interface.

See an example on PRO: https://github.com/moderntribe/events-pro/pull/575

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | There must be a single space after the first parenthesis | [Post_Base.php:201](https://github.com/moderntribe/tribe-common/blob/c8550e7ef194b2e876d4a4761d31d2e7394daf99/src/Tribe/Change_Authority/Post_Base.php#L201)
:x: | There must be a single space before the last parenthesis | [Post_Base.php:201](https://github.com/moderntribe/tribe-common/blob/c8550e7ef194b2e876d4a4761d31d2e7394daf99/src/Tribe/Change_Authority/Post_Base.php#L201)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Change_Authority/Base.php [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	 * @var array
+	 */
+	protected $propagation_conditions = array();
+
+	/**
+	 * Sets the fields the change authority should apply a propagation policy to.
+	 *
+	 * @param array $fields
+	 *
+	 * @return mixed
```
Comment (barryhughes) [inline src/Tribe/Change_Authority/Base.php]: Doesn't return anything

File: src/Tribe/Change_Authority/Base.php [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+		return $result;
+	}
+
+	/**
+	 * Whether a field should be propagated from the source to the destination.
+	 *
+	 * @param mixed  $from The source object or data.
+	 * @param mixed  $to   The destination object or data.
+	 * @param string $type The name of the field that's to be evaluated for propagation.
```
Comment (barryhughes) [inline src/Tribe/Change_Authority/Base.php]: Replace `$type` with `$field` to match the method signature? (Ditto in the docblock for the matching interface definition)

File: src/Tribe/Change_Authority/Base.php [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+		if ( empty( $this->propagation_conditions ) ) {
+			return true;
+		}
+
+		foreach ( $this->propagation_conditions as $callback ) {
+			try {
+				if ( false != call_user_func( $callback, $from, $to, $field ) ) {
+					continue;
+				}
+				throw new RuntimeException( 'Should not propagate' );
```
Comment (barryhughes) [inline src/Tribe/Change_Authority/Base.php]: Why throw the exception here, vs returning false directly?

Comment (lucatume) [inline src/Tribe/Change_Authority/Base.php]: over-engineering. Totally over-engineering.

**Comment by lucatume** [comment]
This too good for another pass.

**Comment by barryhughes** [comment]
👹

**Pull request merged**